zebra-state/src/service/check/anchors.rs

```diff
@@ -103,6 +108,16 @@ pub(crate) fn anchors_refer_to_earlier_treestates(
         // Sapling Spends
         //
         // MUST refer to some earlier block’s final Sapling treestate.
+        //
+        // # Consensus
+        //
+        // > The anchor of each Spend description MUST refer to some earlier
+        // > block’s final Sapling treestate. The anchor is encoded separately
+        // > in each Spend description for v4 transactions, or encoded once and
+        // > shared between all Spend descriptions in a v5
+        // > transaction.
```

Comment:
This consensus rule is naturally implemented in different parts of the codebase. The implementation below effectively deals only with the first sentence of the consensus rule. What's the best way to deal with a situation like this?

I can think of either splitting the consensus rule and mentioning only the relevant part or quoting the whole consensus rule in multiple places in the codebase. None of that seems ideal.

Reply:
We're trying to document consensus rules where they are enforced in the code. Because those locations use the types or functions that implement the rules. So that should limit what we need to document.

But if a rule is enforced in multiple places, I suggest:
1. quote the full rule in both places
2. document the split in both places, with something like:

```ascii
/// This rule is also implemented by [`(type or module link)`].
/// The "(partial rule summary)" check is implemented here.
```

But we have a lot of flexibility here, so let's see how it looks for this specific rule?

Once we have a format we like, let's update the pinned template in Discord. (Or we could put a short consensus rule doc template in `zebra/book/src/dev`).

Reply:
OK, I'll do it as you suggested. Then we can also close #3210.